parquet-column/src/main/java/org/apache/parquet/filter2/predicate/Operators.java

```diff
@@ -345,11 +374,14 @@ public <R> R accept(Visitor<R> visitor) {
     }
 
     /**
-     * Applies a filtering Vistitor to the Contains predicate, traversing any composed And or Or clauses,
-     * and finally delegating to the underlying ColumnFilterPredicate.
+     * Applies a filtering Visitor to the Contains predicate, traversing any composed And or Or clauses,
+     * and finally delegating to the underlying column predicate.
      */
     public abstract <R> R filter(
-        Visitor<R> visitor, BiFunction<R, R, R> andBehavior, BiFunction<R, R, R> orBehavior);
+        Visitor<R> visitor,
+        BiFunction<R, R, R> andBehavior,
+        BiFunction<R, R, R> orBehavior,
+        Function<R, R> notBehavior);
```

Comment:
IIUC, all kinds of filters return all rows if `notBehavior` is found, right?

Reply:
Yes, for example `NotIn` 👍